t/app/model/defecttype.t

```diff
@@ -59,6 +68,44 @@ subtest 'Problem->defect_types behaves correctly' => sub {
     is $problem->defect_types->first->name, $general_defect_type->name, 'Correct defect type is returned for Traffic lights category';
 };
 
+subtest 'by_categories returns all defect types grouped by category' => sub {
+    FixMyStreet::App->model('DB::BodyArea')->create( { body => $oxfordshire, area_id => $area_id } );
+
+    my @contacts = FixMyStreet::DB->resultset('Contact')->not_deleted->search( { body_id => [ $oxfordshire->id ] } )->all;
+    my $defect_types = FixMyStreet::App->model('DB::DefectType')->by_categories($oxfordshire->id, @contacts);
+    my $potholes = decode_json($defect_types->{Potholes});
+    my $traffic_lights = decode_json($defect_types->{'Traffic lights'});
+    my $pavements = decode_json($defect_types->{Pavements});
+
+    is scalar @$potholes, 2, 'Potholes have 2 defect types';
+    is scalar @$traffic_lights, 1, 'Traffic lights have 1 defect type';
+    is scalar @$pavements, 2, 'Pavements have 2 defect types';
+
+    is @$potholes[1]->{extra}->{defect_code}, 123, 'Defect code is present';
+};
+
+subtest 'by_categories returns defect types for an area with multiple bodies' => sub {
+    my $other_body = $mech->create_body_ok('345', 'Some Other Council');
+    FixMyStreet::App->model('DB::BodyArea')->create( { body => $other_body, area_id => $area_id } );
```

Comment:
Area ID is the first argument to `create_body_ok`, so you could put $area_id there and not need this line.